docs/fxa-engineering/fxa-token-types.md

```diff
@@ -0,0 +1,181 @@
+This simplifies client handling of the token since it only has to persist a single value.
+
+# Session Tokens
+
+A `sessionToken` is the Firefox Accounts equivalent of traditional website login session cookie - it represents
+the user who is currently signed in to accounts.firefox.com, and it used to authenticate many calls from the
+FxA front-end website to the FxA back-end API.
+
+Session tokens are an ***FxA Hawk Token*** where the 32-byte string is generated by the server. They can be obtained by:
+* The successful creation of a new account, via the `/account/create` endpoint.
+* A successful login to a new account, via the `/account/login` endpoint.
+* Duplicating an existing `sessionToken`, via the `/session/duplicate` endpoint.
+* Providing the id of a previous `sessionToken` when changing the account password, via the `/password/change/finish`
+  endpoint.
+* Requesting the `https://identity.mozilla.com/tokens/session` scope during an OAuth flow.
+
+Session tokens are intended only for authenticating communication with the FxA servers themselves, so RP servers
+should never encounter a session token in the wild.
+
+On the client side, session tokens are typically handled by web content on accounts.firefox.com while the user
+is interacting with their account. However, web browsers that support signing in with FxA in order to access
+account-enabled browser features, may obtain a session token during the sigin flow and use it to communicate
+directly with the FxA server APIs.
+
+TODO: we should better articulate ownership expections here, around webchannels etc.
+
+# Session Metadata
+
+Since session tokens represent a signed-in user, the server maintains a lot of additional metadata about the user's
+signed-in state. Properties of the session token include:
+
+* The types of authentication performed by the user, and level of confidence in their identity.
+* Timestamp at which authentication last occurred (which can be increased by prompting the user to re-enter their
+  password).
+* Timestamp at which the token was last used (truncated to a few hours granularity to reduce db write load).
+* Details of the user-agent string from the last use of the token.
+
+This information may be used by the FxA server to make authorization decisions, and RPs may request that a user
+be authenticated to a certain level of assurance (such as requiring that the sessionToken be last-authenticated
+within a certain amount of time, or authenticated via 2FA).
+
+## Verified Sessions
+
+As an additional security measure against [credential-stuffing
+attacks](https://blog.mozilla.org/services/2016/04/09/stolen-passwords-used-to-break-into-firefox-accounts/),
+FxA has the notion of a "verified session", which is intended to indicate a higher level of confidence in the
+authenticity of the user.
+
+A session is considered verified if the user did some additional authentication step *in addition* to proving
```

Comment:
This feels very AALy to me, is there room for a model refactor here?

Reply:
Oh my, there is *so much* room for a model refactor here 😆